server/src/board/tile/rail/directioncontrolrailtile.cpp

```diff
@@ -61,6 +61,8 @@ DirectionControlRailTile::DirectionControlRailTile(World& world, std::string_vie
   , setState{*this, "set_state", MethodFlags::ScriptCallable,
       [this](DirectionControlState newState)
       {
+        if(reservedState())
+          return false; // Direction control is currently locked by reserved path
```

Comment:
Just a thought:
If we have `A--(<>)--B` (=both) and there is a reserved path from A to B, should we allow changing it from *both* to *A -> B* ? or the other way around.

As long as it doesn't conflict with the reservation should we allow changing it?

Reply:
Fully agree. Done!

Reply:
Awesome!